src/vs/workbench/parts/tasks/electron-browser/terminalTaskSystem.ts

```diff
@@ -375,7 +375,6 @@ export class TerminalTaskSystem implements ITaskSystem {
 					onData.dispose();
 					onExit.dispose();
 					let key = Task.getMapKey(task);
-					delete this.activeTasks[key];
```

Comment:
I would still keep this since it is the right place to do this and is necessary if the terminal is reused